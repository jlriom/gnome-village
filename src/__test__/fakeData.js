export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
