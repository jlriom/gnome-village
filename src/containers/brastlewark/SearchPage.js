import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import { searchHabitants } from "../../store/actions/searchActions";
import PageLayout from "../../components/layout/PageLayout";
import HabitantItem from "../../components/HabitantItem";

import {
	InputGroup,
	Form,
	FormControl,
	Button,
	Pagination
} from "react-bootstrap";

const SearchPage = ({
	searchCriteria,
	habitantList,
	total,
	error,
	searchHabitants
}) => {
	const [searchByName, setSearchByName] = useState("");
	useEffect(() => {
		const search = {
			...searchCriteria,
			name: searchByName,
			page: 1
		};
		searchHabitants(search);
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, []);

	const handleChange = e => {
		setSearchByName(e.target.value);
	};

	const handleSearch = event => {
		const form = event.currentTarget;
		event.preventDefault();
		if (form.checkValidity() === false) {
			event.stopPropagation();
		} else {
			console.log("searchCriteria", searchCriteria);
			const search = {
				...searchCriteria,
				name: searchByName
			};
			searchHabitants(search);
		}
	};

	const numPages = (searchCriteria, total) => {
		const numPages = Math.trunc(total / searchCriteria.limit);
		return numPages * searchCriteria.limit === total ? numPages : numPages + 1;
	};

	const handlePageChanged = page => {
		const lastPage = numPages(searchCriteria, total);
		if (page < 0) page = 1;
		if (page > lastPage) page = lastPage;
		const search = {
			...searchCriteria,
			page
		};
		searchHabitants(search);
	};
	return (
		<PageLayout>
			<div className='container'>
				<div className='row'>
					<div className='col-sm-3'></div>
					<div className='col-sm-6'>
						<Form onSubmit={handleSearch}>
							<InputGroup>
								<FormControl
									placeholder="Search Brastlewark's habitant"
									aria-label="Search Brastlewark's habitant"
									aria-describedby="Search Brastlewark's habitant"
									value={searchByName}
									onChange={handleChange}
								/>
								<InputGroup.Append>
									<Button variant='secondary' type='submit'>
										Search
									</Button>
								</InputGroup.Append>
							</InputGroup>
						</Form>
					</div>
					<div className='col-sm-3'></div>
				</div>
				<div className='row mt-5'>
					<div className='col center-block'>
						<div
							className='flexBox'
							style={{
								display: "flex",
								flexFlow: "row wrap",
								justifyContent: "center"
							}}>
							{habitantList && (
								<Pagination>
									<Pagination.First onClick={() => handlePageChanged(1)} />
									<Pagination.Prev
										onClick={() => handlePageChanged(searchCriteria.page - 1)}
									/>
									<Pagination.Item>
										{searchCriteria.page} of {numPages(searchCriteria, total)} (
										{total} habitants found)
									</Pagination.Item>
									<Pagination.Next
										onClick={() => handlePageChanged(searchCriteria.page + 1)}
									/>
									<Pagination.Last
										onClick={() =>
											handlePageChanged(numPages(searchCriteria, total))
										}
									/>
								</Pagination>
							)}
						</div>
					</div>
				</div>
			</div>
			{habitantList && (
				<ul>
					{habitantList.map(habitant => (
						<HabitantItem key={habitant.id} habitant={habitant}></HabitantItem>
					))}
				</ul>
			)}
		</PageLayout>
	);
};

const mapStateToProps = state => ({
	searchCriteria: state.search.searchCriteria,
	habitantList: state.search.list,
	total: state.search.total,
	error: state.search.error
});

export default connect(
	mapStateToProps,
	{ searchHabitants }
)(SearchPage);
