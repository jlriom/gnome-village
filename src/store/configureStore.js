import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
