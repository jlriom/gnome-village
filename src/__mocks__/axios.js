import * as fakeData from "./fakeData";

export default {
	get: jest.fn(() =>
		Promise.resolve({
			data: fakeData.habitants
		})
	),
	post: jest.fn(() => Promise.resolve({ data: fakeData.habitant })),
	put: jest.fn(() => Promise.resolve({ data: fakeData.habitant })),
	delete: jest.fn(() => Promise.resolve({ data: fakeData.habitant }))
};
