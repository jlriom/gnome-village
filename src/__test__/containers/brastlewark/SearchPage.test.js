import React from "react";
import SearchPage from "../../../containers/brastlewark/SearchPage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "../../../store/configureStore";
import { habitant, friends } from "../../../__mocks__/fakeData";

const store = configureStore();

const props = {
	id: 0,
	isGuest: false,
	searchCriteria: {},
	habitantList: [habitant],
	total: 1,
	friends: friends,
	searchHabitants: jest.fn(),
	loadFriends: jest.fn(),
	addFriend: jest.fn(),
	removeFriend: jest.fn()
};

describe("<FriendsPage />", () => {
	it("renders ok", () => {
		const tree = renderer.create(
			<Provider store={store}>
				<SearchPage {...props} />
			</Provider>
		);
		expect(tree).toMatchSnapshot();
	});
});
