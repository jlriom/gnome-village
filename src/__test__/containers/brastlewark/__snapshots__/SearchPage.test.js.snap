// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FriendsPage /> renders ok 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-3"
      />
      <div
        className="col-sm-6"
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="input-group"
          >
            <input
              aria-describedby="Search Brastlewark's habitant"
              aria-label="Search Brastlewark's habitant"
              className="form-control"
              onChange={[Function]}
              placeholder="Search Brastlewark's habitant"
              value=""
            />
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-secondary"
                disabled={false}
                type="submit"
              >
                Search
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        className="col-sm-3"
      />
    </div>
    <div
      className="row mt-5"
    >
      <div
        className="col center-block"
      >
        <div
          className="flexBox"
          style={
            Object {
              "display": "flex",
              "flexFlow": "row wrap",
              "justifyContent": "center",
            }
          }
        >
          <ul
            className="pagination"
          >
            <li
              className="page-item"
            >
              <a
                className="page-link"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                <span
                  aria-hidden="true"
                >
                  «
                </span>
                <span
                  className="sr-only"
                >
                  First
                </span>
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                <span
                  aria-hidden="true"
                >
                  ‹
                </span>
                <span
                  className="sr-only"
                >
                  Previous
                </span>
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                1
                 of 
                0
                 (
                0
                 habitants found)
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                <span
                  aria-hidden="true"
                >
                  ›
                </span>
                <span
                  className="sr-only"
                >
                  Next
                </span>
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                <span
                  aria-hidden="true"
                >
                  »
                </span>
                <span
                  className="sr-only"
                >
                  Last
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <ul />
</div>
`;
